theano/gof/destroyhandler.py

```diff
@@ -1026,3 +1027,23 @@ def orderings(self, fgraph):
                         rval[app] = root_clients
 
         return rval
+
+
+class NoOutputFromInplace(Feature):
+
+    def __init__(self):
+        pass
+
+    def validate(self, fgraph):
+        if not hasattr(fgraph, 'destroyers'):
+            return True
+        for out in list(fgraph.outputs):
+            # Validate that the node that produces the output does not produce
+            # it by modifying something else inplace.
+            node = out.owner
+            op = node.op
+            out_idx = node.outputs.index(out)
+            if hasattr(op, 'destroy_map') and out_idx in op.destroy_map.keys():
+                raise InconsistencyError(
+                    "Trying to produce an output ", out,
+                     " by modifying another variable inplace")
```

Comment:
Can you make the error message more clear that this isn't an error by itself, but it is a properties/feature that was requested? Something like:

We where requested(probably for efficiency reason for scan) to not allow outputs of the function to be the result of inplace operation.